--- mm/mmap.c	2026-02-21 17:01:20.788578212 -0700
+++ mm/mmap.c.old	2026-02-21 14:28:17.265305083 -0700
@@ -3870,12 +3870,3 @@
 	return 0;
 }
 subsys_initcall(init_reserve_notifier);
-
-/* we gonna try make our own syscall lol
- * copy user space array to kernel space
- * set all values to 1
- * copy back*/
-SYSCALL_DEFINE(sys_app_helper,int help, int size){
-	printk("We are in the sys_app_helper syscall now!");
-	return 0;
-}
