--- mm/mmap.c	2026-02-23 09:48:52.798412651 -0700
+++ mm/mmap.c.old	2026-02-23 12:26:42.424520067 -0700
@@ -3870,15 +3870,3 @@
 	return 0;
 }
 subsys_initcall(init_reserve_notifier);
-
-SYSCALL_DEFINE2(app_helper,void *,buffer,int,size)
-{
-	int i;
-	char *kernelBuffer = kzalloc(size, GFP_KERNEL);
-	copy_from_user(kernelBuffer,buffer,size);
-	for(i = 0; i < size; i++){
-		kernelBuffer[i] = 1;
-	}
-	copy_to_user(buffer,kernelBuffer,size);
-	return 0;
-}
